DOCUMENTATION.md

```diff
@@ -43,6 +45,153 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+- following the strict order of event writes across streams precludes grouping event processing at the stream level (unless you work in a batched manner). Being able to read ahead, collating events from future batches (while still honoring the batch order with regard to checkpointing etc) afford handlers the ability to process multiple events for a single stream as a group (equally, it allows a handler to determine that a given stream has already reached a particular write position, enabling the ingestion process to discard future redundant reads on the basis as processing catches up) 
+
+### Relationships between data established via projections or reactions
+
+The preceding section lays out why building projections that assume a global order of events and then traversing them serially can be harmful to throughput and make a system harder to operate and/or deploy (and, frequently, the complexity of the processing grows beyond the superficially simple initial logic as you've applied batching and other optimizations).
+
+Nonetheless, there will invariably be parent-child relationships within data and it's processing. When analyzed, these group into at least the following buckets:
+- foreign key relationships reflecting data dependencies that are necessary to serve queries from a read model
+- data that feeds into an overall data processing flow that is reliant on that dependent data being in place for a given piece of processing to yield a correct or meaningful result
+
+Where the parent and child information can live within the same stream, the stream level ordering guarantee can be used to guard against 'child' data being rendered or operated on without its parent being present directly.
+
+#### Explicit Process Managers
+
+In more complex cases, the projection will need to hide or buffer data until such time as the dependency has been fulfilled. At its most general, this is a [Process Manager](https://www.enterpriseintegrationpatterns.com/patterns/messaging/ProcessManager.html). It's important to validate any presumed requirements about being able to render a given read model the instant a child piece of data is ingested into the model; often having a LEFT JOIN in a SELECT exclude the child from a list until such time as the parent has been ingested can be perfectly reasonable regardless of one's desire to have a perfect unified model with all possible foreign keys applied in order to guarantee a watertight model of the entire enterprise (this should be considered alongside other aspects such as whether the projection work is asynchronous, or preemptively carried out prior to reporting completion of a request)
```

Comment:
I really feel for the people trying to maintain foreign key relationships in a SQL read model 😞.

Consider that deletion in that case implies a cascade (and man do we not want a cascade in most cases)

Reply:
Had to mention cascades in https://github.com/jet/propulsion/pull/200/commits/741775c306cde27c4d4d91829a4f60df38b57c17
Could probably benefit from a rewording though...